nucleus/sensorfusion.py

```diff
@@ -0,0 +1,138 @@
+    Point cloud data is assumed to be in ego coordinates.
+    If the point cloud is in World Coordinates, one can provide the inverse pose as the "transform" argument. If this argument
+    is present, the point cloud will be transformed back into ego coordinates
+    Args:
+        points (np.array): Point cloud data represented as (N,3) or (N,4) numpy arrays
+        transform (:class:`Transform`): If in World Coordinate, the transformation used to transform lidar points into ego coordinates
+    '''
+
+    def __init__(self, points: np.array = None, transform: Transform = None):
+        self.points = points
+        if transform is not None:
+            self.points = transform.apply(points)
+
+    def load_pcd(self, filepath: str, transform: Transform = None):
+        '''
+        Takes raw pcd file and reads in as numpy array.
+        Args:
+            filepath (str): Local filepath to pcd file
+            transform (:class:`Transform`): If in world, the transformation used to transform lidar points into ego coordinates
+        '''
+        points = read_pcd(filepath)
+        self.points = points
+        if transform is not None:
+            self.points = transform.apply(points)
+
+
+class CamCalibration:
+    '''
+    CamCalibration solely holds the pose of the camera
+    This CamCalibration will inevitably be transformed by the device_pose
+    Args:
+            extrinsic_matrix (np.array): (4,4) extrinsic transformation matrix representing device_to_lidar
+    '''
+
+    def __init__(self, extrinsic_matrix=None):
+        self.extrinsic_matrix = extrinsic_matrix
+
+    @property
+    def extrinsic_matrix(self):
+        """Camera extrinsic
+        :getter: Return camera's extrinsic matrix (pose.inverse[:3, :4])
+        :setter: pose = Transform(matrix).inverse
+        :type: 3x4 matrix
+        """
+        return self.pose.inverse[:3, :4]
+
+    @extrinsic_matrix.setter
+    def extrinsic_matrix(self, matrix):
+        '''
+        Sets pose as inverse of extrinsic matrix
+        '''
+        self.pose = Transform(matrix).inverse
+
+
+class RawFrame:
+    '''
+    RawFrames are containers for point clouds, image extrinsics, and device pose.
+    These objects most notably are leveraged to transform point clouds and image extrinsic matrices to the world coordinate frame.
+    Args:
+            dev_pose (:class:`Transform`): World Coordinate transformation for frame
+            **kwargs (Dict[str, :class:`RawPointCloud, :class: CameraCalibration`]): Mappings from sensor name
+              to pointcloud and camera calibrations. Each frame of a lidar scene must contain exactly one
+              pointcloud and any number of camera calibrations
+    '''
+    def __init__(self, dev_pose: Transform = None, **kwargs):
+        self.dev_pose = dev_pose
+        self.items = {}
+        for key, value in kwargs.items():
+            if isinstance(value, CamCalibration):
+                self.items[key] = copy.copy(value)
```

Comment:
Otherwise this basically passes by reference. We want to generate a xFrames camera calibrations and this will just continually update the same variable otherwise